modules/ugdk-core/include/ugdk/graphic/mesh.h

```diff
@@ -0,0 +1,91 @@
+
+#ifndef UGDK_GRAPHIC_MESH_H_
+#define UGDK_GRAPHIC_MESH_H_
+
+#include <ugdk/structure/vertex.h>
+#include <ugdk/graphic/canvas.h>
+#include <ugdk/graphic/module.h>
+#include <ugdk/graphic/vertexdata.h>
+#include <ugdk/graphic.h>
+
+#include <glm/vec2.hpp>
+
+#include <vector>
+
+namespace ugdk {
+namespace graphic {
+
+class Canvas;
+
+// TODO:
+// + [x] Add fill methods
+// + [x] Write stream method
+// Remember:
+// + VertexData will be used internally to group buffers
+// + Mesh is used by user for easy access
+
+template <typename Vertex>
+class Mesh {
+  public:
+    /// Alias
+    using VertexArray = std::vector<Vertex>;
+
+    /// Default constructor (an invalid mesh)
+    Mesh();
```

Comment:
What's the use for creating invalid meshes? Maybe at least add a method to check if the mesh is valid?

Reply:
I ended up needing a default constructor, otherwise an example code would become MUCH more complex. I accept suggestions, but will add the checker method.